pow-migration/src/monitor/mod.rs

```diff
@@ -45,12 +47,24 @@ pub async fn get_ready_txns(
     client: &Client,
     validator: String,
     block_window: Range<u32>,
+    pos_genesis_config_hash: &Blake2bHash,
 ) -> Vec<TransactionDetails> {
     if let Ok(transactions) = client.get_transactions_by_address(&validator, 10).await {
+        let genesis_config_hex = pos_genesis_config_hash.to_hex();
+
         let filtered_txns: Vec<TransactionDetails> = transactions
             .into_iter()
             .filter(|txn| {
-                // Here we filter by current epoch
+                // Check if the txn contains extra data and matches our genesis block hash
+                if let Some(txn_data) = &txn.data {
```

Comment:
I'd suggest to factorize the txn data check in a function to avoid code duplication